source/mqtt_client.c

```diff
@@ -15,6 +15,8 @@
 #include "mqtt_client.h"
 #include "io.h"
 
+#include <assert.h>
```

Comment:
assert.h is banned by clang-tidy. Why not use AWS_ASSERT?

Reply:
Cause I didn't put the assert there. I'll go back and update.